### PR TITLE
Fixed iOS framework linking and updated to native SDK 2.3.5

### PR DESCRIPTION
**Pull request by jkasten2**
* Update RCTOneSignal.xcodeproj to link to framework and updated paths to correctly import it.
* Updated to the latest 2.3.5 OneSignal.framework.
* Updated linking steps for the new `RCTOneSignal.xcodeproj`.
* Improved iOS linking steps with screenshots.
* Since `OneSignal.framework` is now included in the project remove references to use Cocoapods or installing it manually.
   - Added manual update as a backup when a fix is needed quickly.

### CODE REVIEW

**Comment by jkasten2** [comment]
This branch can be tested on a new project with the following steps.
```bash
react-native init AwesomeProject
cd AwesomeProject
npm install --save git://github.com/jkasten2/react-native-onesignal#ios_linking_and_header_fix
react-native link react-native-onesignal
```

**Comment by larryranches** [comment]
This PR seems to have resolved the issue. When can we get this merged?

**Comment by jkasten2** [comment]
@larryranches There seems to be an issue still with installing this from `npm`. It doesn't add the symlinks from `OneSignal.framework` that is required for all the pathing to work. This is due to this issue.
https://github.com/npm/npm/issues/7825

I haven't tested this with `yarn` yet by updating the following line in my testing steps.
```
npm install --save git://github.com/jkasten2/react-native-onesignal#ios_linking_and_header_fix
```

If this does work for `yarn` it would leave out the `npm` users. Going to see we can include the full OneSignal native iOS source into this repo instead. This could also lower the number of setup setups even further.

**Comment by larryranches** [comment]
@jkasten2 I can confirm that this does work for yarn 👍

File: README.md [hunk 1]
```diff
@@ -132,38 +134,27 @@ android {
...
 }
 ```
-
-## iOS Installation
+<br>
+## iOS Push Certificate
  * Open OneSignal account here: https://onesignal.com/
  * Follow OneSignal's instructions on generating an iOS Push Certificate: https://documentation.onesignal.com/docs/generate-an-ios-push-certificate
 
-### With CocoaPods
```
Comment (bitsal) [inline README.md]: What about CocoaPods installation version?

**Comment by bitsal** [comment]
What about CocoaPods installation version?

**Comment by jkasten2** [comment]
@larryranches Thanks for confirming.

@bitsal No need to install the native part of the SDK (`OneSignal.framework`) as it is part of the repo in this commit and is linked in with the updated instructions. This also means that for example 3.0.3 of this react SDK will equal 2.3.5 of the iOS native SDK. Updating the react SDK will get you a matching iOS native SDK.

**Comment by jkasten2** [comment]
Closing this pull request as I have implement a fix in another way by using the full iOS native SDK source instead of using a built .framework to fix the `npm` problem.
https://github.com/geektimecoil/react-native-onesignal/pull/187

**Pull request closed without merging**